pkg/intel/metadata/manifest/signature_types.go

```diff
@@ -194,19 +206,32 @@ func (s SignatureRSAASA) String() string {
 }
 
 // Verify implements SignatureDataInterface.
-func (s SignatureRSAASA) Verify(pkIface crypto.PublicKey, signedData []byte) error {
+func (s SignatureRSAASA) Verify(pkIface crypto.PublicKey, hashAlgo Algorithm, signedData []byte) error {
 	pk, ok := pkIface.(*rsa.PublicKey)
 	if !ok {
 		return fmt.Errorf("expected public key of type %T, but received %T", pk, pkIface)
 	}
 
-	h := sha256.New()
+	h, err := hashAlgo.Hash()
+	if err != nil {
+		return fmt.Errorf("invalid hash algorithm: %q", err)
+	}
 	if _, err := h.Write(signedData); err != nil {
 		return fmt.Errorf("unable to hash the data: %w", err)
 	}
 	hash := h.Sum(nil)
 
-	err := rsa.VerifyPKCS1v15(pk, crypto.SHA256, hash, s)
+	var hashfunc crypto.Hash
+	switch hashAlgo {
+	case AlgSHA256:
+		hashfunc = crypto.SHA256
+	case AlgSHA384:
+		hashfunc = crypto.SHA384
+	default:
+		return fmt.Errorf("signature verification for RSAASA only supports SHA256 and SHA384")
+	}
+
+	err = rsa.VerifyPKCS1v15(pk, hashfunc, hash, s)
```

Comment:
Couldn't we use the signature size to detect the hash algo if it is not defined (if it is AlgNULL)?

Reply:
Spec says keysize (in bit)/ 8 = signature size. Looks like there is no way to derive hash algo from that information.